### PR TITLE
Added `EntityRefPresentation#promise` for easier consumption

### PR DESCRIPTION
**Pull request by freben**
Helping out with #24081

Builds a foundation for easier contributions to #20955

### CODE REVIEW

File: plugins/catalog/src/apis/EntityPresentationApi/DefaultEntityPresentationApi.ts [hunk 1]
```diff
@@ -260,47 +260,70 @@ export class DefaultEntityPresentationApi implements EntityPresentationApi {
...
+            entity: newEntity,
+          });
+        }
+        return initialSnapshot;
+      })
+      .catch(() => initialSnapshot);
 
     return {
       snapshot: initialSnapshot,
       update$: observable,
```
Comment (vinzscam) [inline plugins/catalog/src/apis/EntityPresentationApi/DefaultEntityPresentationApi.ts:325]: question not related to this PR: if this observable emits only a single value what's the real point of having an observable? 🤔

File: plugins/catalog-react/api-report.md [hunk 2]
```diff
@@ -460,6 +460,7 @@ export type EntityRefLinksProps<
 
 // @public
 export interface EntityRefPresentation {
+  promise: Promise<EntityRefPresentationSnapshot>;
```
Comment (vinzscam) [inline plugins/catalog-react/api-report.md:463]: maybe?

```suggestion
  promise(): Promise<EntityRefPresentationSnapshot>;
```

File: plugins/catalog/src/apis/EntityPresentationApi/DefaultEntityPresentationApi.ts (hunk 1 above)
Comment (freben) [reply plugins/catalog/src/apis/EntityPresentationApi/DefaultEntityPresentationApi.ts:325]: It does allow for reactive updates in the future too, when the catalog emits events for changed entities :)

But realistically, it's more about how it's sort of a pattern to have synchronous `snapshot` and then optionally an observable. It's what gives you the most power and you can easily consume that pair in a `useObservable`. The observable also gives the ability to NOT emit any change events at all if the code finds that "eh, the snapshot turned out to be perfectly fine". A promise always "emits".

But yeah. In practice this turns out to not be an often needed power.

File: plugins/catalog-react/api-report.md (hunk 2 above)
Comment (freben) [reply plugins/catalog-react/api-report.md:463]: The other ones weren't methods so I didn't make one here either for consistency. I considered maybe making it a getter anyway though behind the scenes for lazy triggering the load+render. May still do that at some point. But people can still consume it then as if it's a regular property.

Comment (vinzscam) [reply plugins/catalog-react/api-report.md:463]: but why do we need to create promises (or observables or others things) if they are not going to be used?

Comment (freben) [reply plugins/catalog-react/api-report.md:463]: You are right in theory. But in practice, there's no code path that does not immediately grab the observable or the promise, so all of the actual heavy work will be triggered no matter what. The only possible savings could be to not create the observable or the promise object itself, saving some minor overhead. But in practice I would argue that it's so minor, that it's not worth worrying about. And remember - a getter in and of itself adds overhead, such that the framework needs to call `Object.defineProperty` and make a function with its closure overhead. Not sure that that's actually measurably cheaper than the Promise creation.

**Review by Rugvip — APPROVED** [review]
Let's :shipit: :+1: 

Was gonna say that it would be nice with a more beneral snapshot + observable -> promise utility, but I realize that's tricky with this particular logic. Only reservations is that this will discourage usage of the observable, but meh, that's fine

**Pull request merged**